service/worker/diagnostics/activities_test.go

```diff
@@ -51,23 +51,14 @@ const (
 func Test__identifyIssues(t *testing.T) {
 	dwtest := testDiagnosticWorkflow(t)
 	actMetadata := failure.FailureMetadata{
-		Identity: "localhost",
-		ActivityScheduled: &types.ActivityTaskScheduledEventAttributes{
-			ActivityID:   "101",
-			ActivityType: &types.ActivityType{Name: "test-activity"},
-			RetryPolicy: &types.RetryPolicy{
-				InitialIntervalInSeconds: 1,
-				MaximumAttempts:          1,
-			},
-		},
-		ActivityStarted: &types.ActivityTaskStartedEventAttributes{
-			Identity: "localhost",
-			Attempt:  0,
-		},
+		Identity:            "localhost",
+		ActivityScheduledID: 2,
+		ActivityStartedID:   3,
```

Comment:
I'd suggest providing activityType together with activityID. This will simplify parsing the data without extra looking into the history.